miden/tests/integration/operations/stack_ops.rs

```diff
@@ -14,17 +197,462 @@ fn swapdw() {
+        let asm_op = format!("dup.{n}");
+        let mut expected_values = test_values.to_vec();
+        let a = expected_values[MIN_STACK_DEPTH-n-1];
+        expected_values.remove(0);
+        expected_values.push(a);
+        expected_values.reverse();
+        build_op_test!(asm_op, &test_values).prop_expect_stack(&expected_values)?;
+    }
+
+    #[test]
+    fn dupw_proptest(test_values in prop::collection::vec(any::<u64>(), MIN_STACK_DEPTH)) {
+        let asm_op = "dupw";
+        let mut expected_values = test_values.to_vec();
+        expected_values.drain(0..WORD_LEN);
+        let mut a = test_values[MIN_STACK_DEPTH-WORD_LEN..].to_vec();
+        expected_values.append(&mut a);
+        expected_values.reverse();
+        build_op_test!(asm_op, &test_values).prop_expect_stack(&expected_values)?;
+    }
+
+    #[test]
+    fn dupwn_proptest(test_values in prop::collection::vec(any::<u64>(), MIN_STACK_DEPTH), n in 0_usize..WORD_LEN) {
+        let asm_op = format!("dupw.{n}");
+        let mut expected_values = test_values.to_vec();
+        expected_values.drain(0..WORD_LEN);
+        let mut a = test_values[MIN_STACK_DEPTH-WORD_LEN*(n+1)..MIN_STACK_DEPTH-WORD_LEN*n].to_vec();
+        expected_values.append(&mut a);
+        expected_values.reverse();
+        build_op_test!(asm_op, &test_values).prop_expect_stack(&expected_values)?;
+    }
+
+    #[test]
+    fn swap_proptest(test_values in prop::collection::vec(any::<u64>(), MIN_STACK_DEPTH)) {
+        let asm_op = "swap";
+        let mut expected_values = test_values.to_vec();
+        expected_values.reverse();
+        expected_values.swap(0, 1);
+        build_op_test!(asm_op, &test_values).prop_expect_stack(&expected_values)?;
+    }
+
+    #[test]
+    fn swapn_proptest(test_values in prop::collection::vec(any::<u64>(), MIN_STACK_DEPTH), n in 1_usize..MIN_STACK_DEPTH) {
+        let asm_op = format!("swap.{n}");
+        let mut expected_values = test_values.to_vec();
+        expected_values.reverse();
+        expected_values.swap(0, n);
+        build_op_test!(asm_op, &test_values).prop_expect_stack(&expected_values)?;
+    }
+
+    #[test]
+    fn swapw_proptest(test_values in prop::collection::vec(any::<u64>(), MIN_STACK_DEPTH)) {
+        let asm_op = "swapw";
+        let mut expected_values = test_values.to_vec();
+        let mut a = expected_values.split_off(WORD_LEN*3);
+        let mut b = expected_values.split_off(WORD_LEN*2);
+        expected_values.append(&mut a);
+        expected_values.append(&mut b);
+        expected_values.reverse();
+        build_op_test!(asm_op, &test_values).prop_expect_stack(&expected_values)?;
+    }
+
+    #[test]
+    fn swapwn_proptest(test_values in prop::collection::vec(any::<u64>(), MIN_STACK_DEPTH), n in 1_usize..WORD_LEN) {
+        let asm_op = format!("swapw.{n}");
+        let mut expected_values = test_values.to_vec();
+        let mut a = expected_values.split_off(WORD_LEN*(MIN_STACK_DEPTH/WORD_LEN-n-1));
```

Comment:
This is a bit hard to read. I would prefer if `WORD_LEN*(MIN_STACK_DEPTH/WORD_LEN-n-1` were defined as a variable with a meaningful name and spacing around the operators.